server/src/api/Services/DatabaseServices/UserDBService.ts

```diff
@@ -88,6 +88,15 @@ export class UserDBService {
         return plainToClass(User, <any>u);
     }
 
+    public getUserByUsername(username: string): User {
+        let u = this.dbService.queryOne<User>({
+            select: ['*'],
```

Comment:
```suggestion
            select: ['id'],
```
shouldn't return * in my opinion because password is returned also (even though it's salted and hashed and also only on the server), only id or all other required fields for items that use this function should be returned. Add fields that are required to be returned later

Reply:
doesn't make a difference since the data never leaves the server. Just selecting the id would also break the type on the query